rpcs3/Emu/System.cpp

```diff
@@ -964,7 +982,97 @@ game_boot_result Emulator::Load(const std::string& title_id, bool add_only, bool
 			}
 
 			m_path_old = m_path;
+			resolve_path_as_vfs_path = true;
+		}
+		else if (m_path.starts_with("%RPCS3_VFS%:"))
+		{
+			m_path = m_path.substr(("%RPCS3_VFS%:"sv).size());
+
+			if (!m_path.empty() && m_path[0] != '/')
+			{
+				// Make valid for VFS
+				m_path.insert(0, "/");
+			}
+
+			if (!argv.empty())
+			{
+				argv[0] = m_path;
+			}
+			else
+			{
+				argv.emplace_back(m_path);
+			}
+
+			resolve_path_as_vfs_path =  true;
+		}
+		else if (m_path.starts_with("%RPCS3_GAMEID%:"))
+		{
+			// Try to boot a game through game ID only
+			m_title_id = m_path.substr(("%RPCS3_GAMEID%:"sv).size());
+			m_title_id = m_title_id.substr(0, m_title_id.find_first_of(fs::delim));
+			std::string tail = m_path.substr(("%RPCS3_GAMEID%:"sv).size() + m_title_id.size());
+
+			if (tail.find_first_not_of(fs::delim) == umax)
+			{
+				// Treat slashes-only trail as if game ID only was provided
+				tail.clear();
+			}
+
+			bool ok = false;
+
+			std::string test_hdd0_path = rpcs3::utils::get_hdd0_dir() + "game/" + m_title_id + "/USRDIR/EBOOT.BIN";
+
+			if (tail.empty() && fs::is_file(test_hdd0_path))
+			{
+				m_path = std::move(test_hdd0_path);
+				ok = true;
+			}
+
+			if (!ok)
+			{
+				if (auto node = games[m_title_id])
+				{
+					std::string path = node.Scalar();
+					
+					if (!tail.empty())
+					{
+						ok = fs::is_file(path + tail + "/USRDIR/EBOOT.BIN");
+
+						if (ok)
+						{
+							path += tail + "/USRDIR/EBOOT.BIN";
+						}
+					}
+					else
+					{
+						ok = fs::is_file(path + "/PS3_GAME/USRDIR/EBOOT.BIN");
+
+						if (ok)
+						{
+							path += "/PS3_GAME/USRDIR/EBOOT.BIN";
+						}
+						else
+						{
+							ok = fs::is_file(path + "/USRDIR/EBOOT.BIN");
 
+							if (ok)
+							{
+								path += "/USRDIR/EBOOT.BIN";
+							}
+						}
+					}
```

Comment:
this can easilly be rewritten to a simple loop with three paths.

Reply:
Fixed.